folio-spring-system-user/src/test/java/org/folio/spring/systemuser/PrepareSystemUserServiceTest.java

```diff
@@ -114,6 +117,25 @@ void shouldAddOnlyNewPermissions() {
       .addPermission(any(), eq(new Permission("inventory-storage.instance.item.post")));
   }
 
+  @Test
+  void updateCredentialsForAnExistingUser() {
+    when(usersClient.query(any())).thenReturn(userExistsResponse());
+    when(permissionsClient.getUserPermissions(any()))
+      .thenReturn(asSinglePage("inventory-storage.instance.item.get"));
+    prepareSystemUser(systemUserProperties());
+    verify(authnClient).deleteCredentials(any());
+    verify(authnClient).saveCredentials(any());
+  }
+
+  @Test
+  void donNotupdateCredentialsForAnExistingUserWithoutSystemUserPassword() {
+    when(usersClient.query(any())).thenReturn(userExistsResponse());
+    when(permissionsClient.getUserPermissions(any()))
+      .thenReturn(asSinglePage("inventory-storage.instance.item.get"));
+    prepareSystemUser(systemUserPropertiesWithoutPassword());
+    verifyNoInteractions(authnClient);
```

Comment:
```suggestion
      verify(authnClient, never()).deleteCredentials(any());
      verify(authnClient, never()).saveCredentials(any());
```
Might be more symmetrical with the positive test case.

Reply:
Done